lib/src/objectid/objectid.dart

```diff
@@ -38,16 +38,16 @@ class ObjectId {
   /// ObjectId bytes.
   Uint8List get bytes => _bytes;
 
-  DateTime _timestamp;
+  DateTime? _timestamp;
 
   /// Returns the generation date (accurate up to the second) that this
   /// ObjectId was generated.
   DateTime get timestamp {
-    if (_timestamp != null) return _timestamp;
+    if (_timestamp != null) return _timestamp!;
 
     var secondsSinceEpoch = 0;
     for (var x = 3, y = 0; x >= 0; x--, y++) {
-      secondsSinceEpoch += _bytes[x] * math.pow(256, y);
+      secondsSinceEpoch += (_bytes[x] * math.pow(256, y)).round();
```

Comment:
@rknell, 
What's the purpose of this `round()` methods here?

Reply:
I see. Replaced it with `toInt` as `num` returned from `math.pow(256, y)` will always be of `int` type.
Also, `as int` was slower in the benchmark 🤔

Reply:
Ahh didn't realise it would be slower! Yeah it was just to alleviate the issue. Didn't have much to do with null safety I suppose but may have only surfaced because of the new dart version bump.